README.md

```diff
@@ -6,12 +6,20 @@
 
 # Project Goals
 
-The goal of the `Taped.jl` project is to produce a reverse-mode AD package, written entirely in Julia, which improves over both ReverseDiff.jl and Zygote.jl in several ways:
+The goal of the `Taped.jl` project is to produce a reverse-mode AD package, written entirely in Julia, which improves over both `ReverseDiff.jl` and `Zygote.jl` in several ways:
 1. performance,
 1. correctness / scope of testing,
 1. coverage of language features.
 
-The most notable feature that we improve on over ReverseDiff and Zygote / ChainRules is support for mutation (writing to arrays, modifying fields of `mutable struct`s, etc), which is arguably the core limitation of these two packages, and has been the elephant-in-the-room of reverse-mode AD in Julia for years.
+Some notable features that we try to improve on over ReverseDiff and Zygote / ChainRules are 
+
+- optimise performance, usability and robustness while balancing generalizability
+- guaranteed safety against dynamic control flows when using cached and compiled tapes
+- compatibility with more array types, e.g. GPU arrays
+- builtin support for mutation (writing to arrays, modifying fields of `mutable struct`s, etc), which is a shared limitation of `ReverseDiff.jl` and `Zygote.jl`, and has been the elephant-in-the-room of reverse-mode AD in Julia for years
+
+`Taped.jl` inherits many features and characteristics from `ReverseDiff.jl` and aims to be a drop-in replacement. However, compared to `ReverseDiff.jl`'s operator-overloading approach to tracing, we adopt an IR-based tracking based on [Umlaut](https://github.com/dfdx/Umlaut.jl). This IR-based tracing mechanism allows us to avoid the need to rewrite functions with generic type signatures (similar to [`Cassette.jl`](https://github.com/JuliaLabs/Cassette.jl)). 
+
```

Comment:
I'm not in favour of this re-phrasing.

The reason that I only went with mutation is that mutation is missing in _both_ ReverseDiff and Zygote. This is not true of the other points -- the dynamic control flow point is only relevant to ReverseDiff, as is the compatibility with more array types point (Zygote is fine with whatever array you like).

Additionally, my feeling is that the "optimise performance, usability..." point is too vague to be helpful. If 10 different people read it, you'll get 10 different impressions of what it means. My feeling is that the goals at the top of the readme better capture what this sentence states.

I would prefer it if you removed this proposed change.

Reply:
I will include a careful comparison of Taped vs ReverseDiff / Zygote / ChainRules / Enzyme in the docs, so I can add a sentence here along the lines of `An in-depth comparison can be found in the [docs](link to dicussion)` when they're live.

Reply:
I don't know what you mean by a "balanced message" -- could you please be more specific?

Does the colleague you mention have much experience with AD in Julia? I ask, because the readme is not currently intended for a general audience -- it's aimed at people who have some experience in the development of AD systems in Julia, and are likely to already have an appreciation of the problems that lack of mutation support causes. I'm targetting this group of people beacuse they are the ones that we are planning to share this with in the immediate future. If your colleague falls into this group, then I'll happily add further clarification / more context to expain why this is a central issue.

Reply:
I'm also happy to add a note at the top of the readme that clarifies who the (current) intended audience of the package is.